### ISSUE TITLE
sender variable set to channel destination, not sender in PRIVMSG

### ISSUE THREAD

**Issue by jgrar** [mischief/glenda#7]
For a PRIVMSG shouldn't the sender be the message author, not the message desitination? Here's how it's set currently: https://github.com/mischief/glenda/blob/3146fee2798cd9170fc3706906790eada0c796a2/main.go#L85-89

It would be nice to know, from the hook function's perspective, things like "*this is who authored the line*" and "*this is where to send messages back*"

**Comment by mischief** [comment]
yes. it should be fixed. all of the irc package's crap should be wrapped up in a message context struct of our own. i changed this to remove redundant code in all the modules that were figuring out who to reply to on their own.

**Comment by jgrar** [comment]
I was wondering; should we just return an output string from the hook function and send it back to the appropriate target from outside of it? I can show you what I mean in a PR. I've got some time to do it now.
